File: Autos-stock-manager-services/src/main/java/org/autostock/dtos/DepenseDto.java
package org.autostock.dtos;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;
import org.autostock.enums.CategorieDepense;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
public class DepenseDto {
    private Long id;

    private Long voitureId;
    private Long entretienId; // nullable
    private Long documentId;  // nullable

    private CategorieDepense categorie;
    private BigDecimal montant;

    @JsonFormat(pattern = "yyyy-MM-dd['T'HH:mm[:ss]]")
    private LocalDateTime dateDepense;

    private String description;
    private Long fournisseurId;


}

File: autos-stock-client/src/app/models/DepenseDto.ts
import {DepenseCreateDto} from "./DepenseCreateDto";

export interface DepenseDto {
  id: number;
  voitureId: number;
  entretienId?: number | null;
  documentId?: number | null;

  fournisseurId?: number | null;
  fournisseurNom?: string | null;

  categorie: string;
  montant: number;
  dateDepense: string; // YYYY-MM-DD

  description?: string | null;
  kilometrage?: number | null;
}
